gossip/emitter/throttling/dominant_set_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2025 Sonic Operations Ltd
+// This file is part of the Sonic Client
+//
+// Sonic is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Sonic is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with Sonic. If not, see <http://www.gnu.org/licenses/>.
+
+package throttling
+
+import (
+	"math/rand"
+	"slices"
+	"testing"
+
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
+	"github.com/Fantom-foundation/lachesis-base/inter/pos"
+	"github.com/stretchr/testify/require"
+)
+
+func TestComputeDominantSet_CanIdentifyWhenStakeDistributionIsDominated(t *testing.T) {
+
+	tests := map[string]struct {
+		stakes            []int64
+		expectedDominated bool
+		expectedSet       []idx.ValidatorID
+	}{
+		"no validators": {
+			stakes:            nil,
+			expectedDominated: false,
+		},
+		"single validator below threshold": {
```

Comment:
nit: comment somewhere near the cases that the threshold is actually a given constant for the whole test. Perhaps define it above `tests` and in test cases declaration use `threshold/2` or `threshold-1`, etc.